Coffee to REAL js

ramezrafla opened the pull request:
- Note the change in API
- The actual SubsCache file is used in production directly in our app
- package.js was updated - but not tested (testers welcome), however, no expectation of isses

afruth commented on package.js:
```diff
@@ -17,9 +17,9 @@ Package.onUse(function(api) {
   ], ['client', 'server']);
```
Is coffeescript still needed as a dependency?

ramezrafla replied:
Removed

afruth commented on src/SubsCache.js:
```diff
@@ -0,0 +1,223 @@
+
+  this.subscribeFor = function(expireTime, ...args) {
+    var hash = EJSON.stringify(withoutCallbacks(args));
+
```
Is SSR compatibility broken here? In the original SSR was specifically handled.

ramezrafla replied:
Why would SSR compatibility be broken in these lines? Guidance appreciated, don't use SSR.

ramezrafla commented:
Thanks @afruth 
All issues fixed -- didn't know that the server export was for SSR (don't use that). Thanks!

ramezrafla commented on src/SubsCache.js:
```diff
@@ -0,0 +1,223 @@
+  }
+
+  this.subscribe = function(...args) {
+    Logger.log('SubsCache - subscribe',args);
```
Replaced with console

paranoico commented:
@ramezrafla Your PR has been merged, it was supposed that we was going to use another branch.

But, after a bit thinking i see it OK to merge on master, but keeping atmosphere package the same until we are sure there is no issues.

Is it OK by you?

ramezrafla commented:
Agreed!
Thanks for the merge. Waiting for @afruth and others to review before labelling prod ready

afruth commented:
I'll integrate it in one of my apps today, but it will get in production in two weeks so I can't confirm if it's ready or not. Looks pretty close to the original though. Nice job :)




On Wed, Feb 1, 2017 at 1:24 AM +0200, "Ramez Rafla" <notifications@github.com> wrote:












Agreed!

Thanks for the merge. Waiting for @afruth and others to review before labelling prod ready



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

paranoico commented:
Hello @ramezrafla 
We have had several problems with new coding in JS.
Months ago we thought it was Ground:db we use in conjunction in our biggest Cordova App, but today we found that the problems arises with new JS code:
- We got following errors that do not arise using CoffeScript version and those make our app restart:
-- Uncaught (in promise) [object DOMError]
-- WebSocket is already in CLOSING or CLOSED state
-- Sometimes we have an Alert saying: The connection to server was unsuccessful.
- On the other hand, JS code seems to be slower and not responding for minutes.

When restarting with CoffeScript version, problems seems to disappear.
Have you seen such problems on your app?
Thanks in advance.

ramezrafla commented:
Hi @paranoico 
On the contrary, it's working super-great. We did make some improvements, but we moved it out of a package into core (there was also a minor bug we fixed). I could push onto a gist if you like

paranoico commented:
@ramezrafla
Yes please, it would be great to have a copy if you can. Thanks a lot.

paranoico commented:
Hello @ramezrafla ,
We finally have discovered that all raised errors on Cordova comes from Appcache, not from this component.
Strange enough to make me think it was the new JS version since only when using new version the errors appear.
It seems that Appcache gets alot bigger when using JS than when using CoffeeScript.
I am going to update Atmosphere later today with new version, may be it will be have then a smaller size and Appcache will get smaller on our app.
Thanks a lot for your time.

ramezrafla commented:
I don't user AppCache, I use Service Workers. Take a look at NitroBay's sw.js
On Cordova you don't need to worry about it as it stores all files to be rendered by a built-in web server.

paranoico commented:
Thanks again, we will check it out.
